Code/Framework/AzCore/AzCore/Math/Ray.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/RTTI/TypeInfoSimple.h>
+
+namespace AZ
+{
+    class LineSegment;
+
+    class Ray
+    {
+    public:
+        AZ_TYPE_INFO(Ray, "{0301a872-5bea-4563-8070-85ed243cb57c}");
+
+        Ray() = default;
+        Ray(const AZ::Vector3& origin, const AZ::Vector3& direction);
+        
+        static Ray CreateFromLineSegment(const LineSegment& segment);
+
+        const AZ::Vector3& GetOrigin() const;
+        const AZ::Vector3& GetDirection() const;
+    private:
+        AZ::Vector3 m_origin;
+        AZ::Vector3 m_direction;
```

Comment:
question: Do we want to have `SetOrigin` and `SetDirection` too? (there may be cases where you do want to mutate them) and if this is the case, maybe we should just make the members public as having get/setters for this is usually just noise and makes expressions like `ray.m_origin += delta` (I may be in the minority about this though 😅)

Reply:
might want to be consistent with Sphere, Obb? would that be much of a problem?

Reply:
Hmm yeah consistency likely wins out here for now, so maybe the setters are the way to go (others may want to weigh-in though)